src/event_manager/subscribers.rs

```diff
@@ -0,0 +1,51 @@
+use super::SubscriberId;
+use std::collections::HashMap;
+
+// Internal structure used to keep the set of subscribers registered with an EventManger.
+// This structure is a thin wrapper over a `HashMap` in which the keys are uniquely
+// generated when calling `add`.
+pub(crate) struct Subscribers<T> {
+    // The key is the unique id of the subscriber and the entry is the `Subscriber`.
+    subscribers: HashMap<SubscriberId, T>,
+    // We are generating the unique ids by incrementing this counter for each added subscriber,
+    // and rely on the extremely large value range of u128 to ensure each value is effectively
+    // unique over the runtime of any VMM.
+    next_id: u128,
```

Comment:
Why u128 is preferred instead of u64? I think u64 should be bigger enough and more efficient than u128.

Reply:
I'm fine with using u64 instead, @alexandruag what do you think?

Reply:
Yes let's use u64.